theseus/src/api/profile.rs

```diff
@@ -480,6 +481,50 @@ pub async fn remove_project(
     }
 }
 
+/// Exports the profile to a Modrinth-formatted .mrpack file
+// Version ID of uploaded version (ie 1.1.5), not the unique identifying ID of the version (nvrqJg44)
+#[tracing::instrument]
+pub async fn export_mrpack(
+    profile_path: &Path,
+    export_path: PathBuf,
+    included_overrides: Vec<String>, // which folders to include in the overrides
+    version_id: Option<String>,
+) -> crate::Result<()> {
+    let state = State::get().await?;
+    let io_semaphore = state.io_semaphore.0.read().await;
+    let permit: tokio::sync::SemaphorePermit = io_semaphore.acquire().await?;
+    let profile = get(profile_path, None).await?.ok_or_else(|| {
+        crate::ErrorKind::OtherError(format!(
+            "Tried to export a nonexistent or unloaded profile at path {}!",
+            profile_path.display()
+        ))
+    })?;
```

Comment:
this should probably call sync projects on the profile before everything is exported.